src/utils/PlayerManager.java

```diff
@@ -1,130 +1,137 @@
+{
+    private static volatile PlayerManager instance = null;
+    private static volatile ArrayList<Player> playerList = null;
+    private static volatile Player player = null;
+	
+    /**
+     * Private PlayerManager constructor.
+     */
+	private PlayerManager()
+	{
+        /**
+         *  This checks to ensure no other instance is created using Reflection API
+         */
+        if (instance != null)
+        {
+        	System.out.println("Exception Handled: Attempt to create new Instance using Reflection API");
+        }
+        else
+        {
+        	playerList = new ArrayList<Player>();
+        	player = new Player();
+        	System.out.println("[INFO]: Player Manager Initialized.");
+        }
+	}
+	
+
+	/**
+	 * Returns the singleton instance of PlayerManager class.
+	 * @return PlayerManager instance.
+	 */
+    public static PlayerManager getInstance() 
+    {
+        if (instance == null) 
+        { 
+        	/**
+        	 *  This is a thread-safe check to ensure another thread can't initialize another MessageReceiver class.
+        	 */
+            synchronized (PlayerManager.class) 
+            {
+                if (instance == null)
+                {
+                	instance = new PlayerManager();
+                }
+                
+            }
+        }
+
+        return instance;
+    }
+    
+	/**
+	 * Initializes the player object.
+	 * This function sends a notification message to the server and will in return
+	 * receive a unique player ID.
+	 * @param name Name of player.
+	 */
+    public static void initializePlayer(String name)
+    {
+    	PlayerStatusMessage message = new PlayerStatusMessage();
+    	message.setName(name);
+    	message.setType(Action.INIT);
+    	System.out.println("Client Action: Initializing new Client: " + name);
+    	
+    	ClientMessageReceiver.sendMessage(message);
+    }
+    
+    /**
+     * Sets the Player Object.
+     * This function is called when the server sends back a message containing the players unique ID.
+     * @param obj Player Object received by server.
+     */
+    public static void storInitPlayer(Player obj)
+    {
+    	if (obj instanceof Player)
+    	{
+    		PlayerManager.player = obj;
+    	}
+    }
+
+	public static Player getPlayer() 
+	{
+		return player;
+	}
+	
+	public static void setPlayer(Player obj)
+	{
+		player = obj;                
+	}
+	
+	public synchronized static void addNewPlayer(Player player)
+	{       
+                //sort the list by ID
+		playerList.add(player);               
+		System.out.println("New Player: " + player);          
+	}
+	
+	public synchronized static void removePlayer(Player player)
+	{
+		playerList.remove(player);
+	}
+	
+	public static int playerCount()
+	{
+                System.out.println(playerList);
+		return playerList.size() + 2;
+                
+	}
+	
+	public synchronized static Player getOtherPlayer(int playerId)
+	{
+                for (Player currentPlayer : playerList)

```

Comment:
This seems like it will work just fine but I thought the "get" method already returns the Player object matching the playerId? Could be wrong though, either way, don't see any issue with the change

Reply:
the issue was that player's weren't stored in the list by their player id. So player 2 would have an ID of 1, but would be stored in index 0 on Player 1s list. the original method used the player ID and index interchangeably which was causing the get to return the wrong player sometimes.

I changed it to iterate over the list and grab the corresponding ID instead, so we don't have to worry about how they're stored in the list.